dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESReadOnlyMonitor.java

```diff
@@ -0,0 +1,136 @@
+
+    private ESReadOnlyMonitor(final SystemMessageEventUtil systemMessageEventUtil, final RoleAPI roleAPI) {
+        super();
+        this.systemMessageEventUtil = systemMessageEventUtil;
+        this.roleAPI = roleAPI;
+        started.set(false);
+    }
+
+    private ESReadOnlyMonitor() {
+        this(SystemMessageEventUtil.getInstance(), APILocator.getRoleAPI());
+    }
+
+    // Inner class to provide instance of class
+    private static class Singleton
+    {
+        private static final ESReadOnlyMonitor INSTANCE = new ESReadOnlyMonitor();
+    }
+
+    @VisibleForTesting
+    public static ESReadOnlyMonitor getInstance(
+            final SystemMessageEventUtil systemMessageEventUtil,
+            final RoleAPI roleAPI)
+    {
+        return new ESReadOnlyMonitor(systemMessageEventUtil, roleAPI);
+    }
+    public static ESReadOnlyMonitor getInstance()
+    {
+        return Singleton.INSTANCE;
+    }
+
+    public void start(final ReindexEntry reindexEntry, final String cause){
+        if (started.compareAndSet(false, true)) {
+            Logger.error(this.getClass(), "Reindex failed for :" + reindexEntry + " because " + cause);
+
+            if (ESIndexUtil.isEitherLiveOrWokingIndicesReadOnly()) {
+                ReindexThread.setCurrentIndexReadOnly(true);
+                sendMessage("es.index.read.only.message");
+                startMonitor();
+            } else {
+                started.set(false);
+            }
+        }
+    }
+
+    private void sendMessage(final String messageKey) {
+        try {
+            final Role adminRole = roleAPI.loadCMSAdminRole();
+            final List<String> usersId = roleAPI.findUsersForRole(adminRole)
+                    .stream()
+                    .map(user -> user.getUserId())
+                    .collect(Collectors.toList());
+
+            final SystemMessageBuilder message = new SystemMessageBuilder()
+                    .setMessage(LanguageUtil.get(messageKey))
+                    .setSeverity(MessageSeverity.ERROR)
+                    .setType(MessageType.SIMPLE_MESSAGE)
+                    .setLife(TimeUnit.SECONDS.toMillis(5));
+
+            systemMessageEventUtil.pushMessage(message.create(), usersId);
+        } catch (final LanguageException | DotDataException | DotSecurityException e) {
+            Logger.warn(ESReadOnlyMonitor.class, () -> e.getMessage());
+        }
+    }
+
+    private void putCurrentIndicesToWriteMode() {
+        try {
+            Logger.debug(this.getClass(), () -> "Trying to set the current indices to Write mode");
+            ESIndexUtil.setLiveAndWorkingIndicesToWriteMode();
+            sendMessage("es.index.write.allow.message");
+            ReindexThread.setCurrentIndexReadOnly(false);
+
+            this.stop();
+        } catch (final ElasticsearchResponseException e) {
+            Logger.info(ESReadOnlyMonitor.class, ()  -> e.getMessage());
+        }
+    }
+
+    private synchronized void startMonitor() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use block level rather than method level synchronization](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5642172)